action.yaml

```diff
@@ -0,0 +1,196 @@
+      The file must not have the `name` property.
+    required: false
+  recipe-directory:
+    description: The directory containing the recipe file.
+    required: true
+  standalone-location:
+    description: The location of the standalone binary executable.
+    required: false
+
+  # secrets needed by constructor
+  azure-signtool-key-vault-client-id:
+    description: Client ID of the Azure key vault.
+    required: false
+  azure-signtool-key-vault-certificate:
+    description: Certificate of the Azure key vault.
+    required: false
+  azure-signtool-key-vault-secret:
+    description: Secret for the Azure key vault.
+    required: false
+  azure-signtool-key-vault-tenant-id:
+    description: Tenant ID of the Azure key vault.
+    required: false
+  azure-signtool-key-vault-url:
+    description: URL to for the Azure key vault.
+    required: false
+  constructor-pfx-certificate-password:
+    description: Windows signing password.
+    required: false
+
+outputs:
+  artifacts-directory:
+    description: Directory containing the artifacts of the constructor run.
+    value: ${{ steps.outputs.outputs.artifacts-directory }}
+
+runs:
+  using: composite
+  steps:
+    - name: Validate input
+      run: |
+        if [[ -z "${{ inputs.environment-yaml-file }}" ]] && [[ -z "${{ inputs.environment-yaml-string }}" ]]; then
+          echo "::error::\`environment-yaml-file\` or \`environment-yaml-string\` must be set."
+          exit 1
+        fi
+      shell: bash
+
+    - name: Set up workspace
+      id: workspace
+      run: |
+        UNAME="$(uname)"
+        CONSTRUCTOR_WORKSPACE="${{ runner.temp }}/_constructor_workdir"
+        if [[ "${UNAME}" == MINGW* ]]; then
+            CONSTRUCTOR_WORKSPACE=$(cygpath "${CONSTRUCTOR_WORKSPACE}")
+        fi
+        mkdir -p "${CONSTRUCTOR_WORKSPACE}"
+        chmod 777 "${CONSTRUCTOR_WORKSPACE}"
+        echo "constructor-workspace=${CONSTRUCTOR_WORKSPACE}" >> ${GITHUB_OUTPUT}
+        # To avoid permission issues with GITHUB_OUTPUTS inside Docker images,
+        # create an outputs file that can be dumped into the output in a separate step.
+        echo "outputs-file=${CONSTRUCTOR_WORKSPACE}/outputs.txt" >> ${GITHUB_OUTPUT}
+      shell: bash
+
+    - name: Create environment.yaml file
+      if: ${{ !inputs.environment-yaml-file }}
+      id: environment-str
+      env:
+        ENVIRONMENT_FILE: ${{ steps.workspace.outputs.constructor-workspace }}/environment.yaml
+      run: |
+        echo "${{ inputs.environment-yaml-string }}" > "${ENVIRONMENT_FILE}"
+        echo "environment-yaml-file=${ENVIRONMENT_FILE}" >> ${GITHUB_OUTPUT}
+      shell: bash
+
+    - name: Set up QEMU
+      if: ${{ inputs.container-image && inputs.container-arch }}
+      uses: docker/setup-qemu-action@53851d14592bedcffcf25ea515637cff71ef929a     # v3.3.0
+      with:
+        # https://github.com/docker/setup-qemu-action/issues/188#issuecomment-2604322104
+        image: tonistiigi/binfmt:qemu-v8.1.5
```

Comment:
Can we use the hash here, instead of the tag? Trying to minimize supply chain risks.